bimmer_connected/api/utils.py

```diff
@@ -162,18 +163,25 @@ def generate_cn_nonce(username: str) -> str:
     k2 = key[8:]
     i2 = iv[8:]
 
-    sha256_hex = SHA256.new((k1 + i1 + "u3.1.0" + k2 + i2).encode()).hexdigest()
+    if username is None:
+        username = ""
+
+    sha256_hex = SHA256.new((k2 + i1 + "u3.6.1" + username[-4:] + k1 + i2).encode()).hexdigest()
```

Comment:
Would it make sense to get the version from `APP_VERSIONS[regions.CHINA]`? Would of course require to strip the build number.

Reply:
I have thought about this issue before, but Regions.CHINA still needs one more operation to remove the build number `(23634)`. Currently, it seems that each time the China region cannot log in, it not only requires changing the version number but also adjusting the nonce algorithm logic. Therefore, I treat `u3.6.1` as a hardcoded value for the algorithm.